.github/workflows/tests.yml

```diff
@@ -27,13 +27,49 @@ jobs:
       - name: Rebase to current master
         run: git rebase origin/master
 
+      - name: Prepare
+        run: |
+          # Put the /secrets in place which we'll mount to the tasks-container
+          mkdir -p ~/secrets
+          echo '${{ secrets.COCKPITUOUS_TOKEN }}' > ~/.config/github-token
+          echo '${{ secrets.SSH_KNOWN_HOSTS }}' > ~/secrets/known_hosts
+          echo '${{ secrets.SINK_SSH_CONFIG }}' > ~/secrets/ssh-config
+          echo '${{ secrets.SINK_ID_RSA }}' > ~/secrets/id_rsa
+          echo '${{ secrets.SINK_ID_RSA_PUB }}' > ~/secrets/id_rsa.pub
+          chmod 400 ~/secrets/*
+          # change permissions to user in cockpit-tasks container
+          sudo chown 1111:1111 ~/secrets/*
+
       - name: Build and test tasks container if it changed
         run: |
           changes=$(git diff --name-only origin/master..HEAD -- tasks/)
           if [ -n "${changes}" ]; then
               # Run podman as root, as podman is missing slirp4netns by default, and does not have overlayfs by default
               sudo make tasks-container
-              # This does not do much yet, just iterate to the AMQP queue 30 times and exit
-              # TODO: Inject a workload that does not need /dev/kvm
-              sudo tasks/run-local.sh
+
+              # Start the amqp/tasks pod in the background and supply working task secrets
+              sudo tasks/run-local.sh -d -s ~/secrets -t ~/.config/github-token
+
+              # trigger a test
+              PRN=$(echo "$GITHUB_REF" | cut -f3 -d '/')
+              # wait until bots checkout
+              sudo podman exec -i cockpituous-tasks sh -exc "
+              until [ -f cockpit-project/bots/tests-trigger ]; do
+                  sleep 5;
+              done;
+              cd cockpit-project/bots;
+              ./tests-trigger -f --repo cockpit-project/cockpituous $PRN unit-tests;
+              until (./tests-scan --repo cockpit-project/cockpituous -p $PRN -dv 2>/dev/null | grep -q 'unit-tests'); do
+                 ./tests-scan --repo cockpit-project/cockpituous -p $PRN -dv;
+                  echo waiting until the status is visible;
+                  sleep 10;
+              done;
+              ./tests-scan -p $PRN --amqp 'localhost:5671' --repo cockpit-project/cockpituous;
+              ./inspect-queue;"
```

Comment:
Can we rely on that being *that* synchronous? Then again it's not doing any assertions about it, so that's just for debugging? Should it loop until the test is queued? That may be useful for directly seeing *why* a test is failing (instead of idly looping on an empty queue later)

Reply:
It's just for debugging, the queuing is synchronous, there's of course a small chance the tasks container will grab it before inspect-queue can grab/release it :)